bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -1384,12 +1384,6 @@ void ReplicaImp::onMessage(CheckpointMsg *msg) {
   if (askForStateTransfer) {
     LOG_INFO_F(GL, "call to startCollectingState()");
 
-    if (ps_) {
```

Comment:
If we are no longer using `setFetchingState` we should remove it and `getFetchingState` completely in a future commit.

```
andrewstone@ubuntu:~/concord-bft$ grep getFetchingState bftengine/src/bftengine/*
bftengine/src/bftengine/DebugPersistentStorage.cpp:bool DebugPersistentStorage::getFetchingState() {
bftengine/src/bftengine/DebugPersistentStorage.hpp:  bool getFetchingState() override;
bftengine/src/bftengine/PersistentStorage.hpp:  virtual bool getFetchingState() = 0;
bftengine/src/bftengine/PersistentStorageImp.cpp:bool PersistentStorageImp::getFetchingState() {
bftengine/src/bftengine/PersistentStorageImp.hpp:  bool getFetchingState() override;
andrewstone@ubuntu:~/concord-bft$ grep setFetchingState bftengine/src/bftengine/*
bftengine/src/bftengine/DebugPersistentStorage.cpp:void DebugPersistentStorage::setFetchingState(bool state) {
bftengine/src/bftengine/DebugPersistentStorage.hpp:  void setFetchingState(bool state) override;
bftengine/src/bftengine/PersistentStorage.hpp:  virtual void setFetchingState(bool state) = 0;
bftengine/src/bftengine/PersistentStorageImp.cpp:  setFetchingStateInternal(fetchingState_);
bftengine/src/bftengine/PersistentStorageImp.cpp:void PersistentStorageImp::setFetchingStateInternal(uint8_t state) {
bftengine/src/bftengine/PersistentStorageImp.cpp:void PersistentStorageImp::setFetchingState(bool state) {
bftengine/src/bftengine/PersistentStorageImp.cpp:  setFetchingStateInternal(state);
bftengine/src/bftengine/PersistentStorageImp.hpp:  void setFetchingState(bool state) override;
bftengine/src/bftengine/PersistentStorageImp.hpp:  void setFetchingStateInternal(uint8_t state);
bftengine/src/bftengine/ReplicaImp.cpp:      ps_->setFetchingState(true);
bftengine/src/bftengine/ReplicaImp.cpp:    ps_->setFetchingState(false);
bftengine/src/bftengine/ReplicaImp.cpp:      ps_->setFetchingState(true);
```

Reply:
I agree